akirachix/Taishi-Frontend
# Added Sidebar

## Engineers Checklist
## What this PR does?
This PR implements a  sidebar component for our Next.js application using Tailwind CSS. The sidebar includes navigation items with icons, active state highlighting, and a logo.

## What is this?
This is a Sidebar component for our Themis.ai application.
## Why this?
We need a consistent and visually appealing navigation sidebar that highlights the current active page and provides easy access to key sections of our application.
## How are you doing this?

Implementing a React functional component using Next.js and Tailwind CSS
Using Lucide React for icons
Implementing custom hook (useSidebar) for managing active path and navigation
Applying responsive design principles for various screen sizes




## List of updates for features, enhancements, or bugfixes

Created a Sidebar component with a logo and navigation items
Implemented active state highlighting for the current page
Added underline effect for navigation items
Ensured responsiveness and consistent spacing
Fixed bug related to null icon rendering for the Profile item



## Type of change

 New feature (non-breaking change which adds functionality)
 Bug fix (non-breaking change which fixes an issue)
 Improvement

## How can it be tested?

Clone the repository and checkout this branch
Run npm install to install dependencies
Start the development server with npm run dev
Navigate to different pages and observe the sidebar behavior:

Check if the active item is highlighted in yellow (#F99D15)
Verify that the underline appears correctly for each item
Ensure the Profile item appears without an icon but maintains correct spacing


Test responsiveness by resizing the browser window

Screenshots:
![Screenshot from 2024-09-18 12-15-52](https://github.com/user-attachments/assets/25098eed-9415-41bd-97d0-8a26b44dba23)

Where is it documented?

### JIRA ticket: https://lavenderanyango.atlassian.net/browse/TG-140

Note: This PR implements the sidebar component as discussed. Further adjustments to colors, spacing, or functionality can be made based on additional feedback or requirements.
